src/bespokelabs/curator/request_processor/base_request_processor.py

```diff
@@ -275,8 +288,19 @@ def create_dataset_files(
             try:
                 writer.finalize()
             except SchemaInferenceError as e:
+                os.remove(dataset_file)
+                raise ValueError(
+                    "Arrow writer is complaining about the schema: likely your `parse_func` returned only None objects. Please fix your `parse_func` to return valid rows and re-run. The dataset will be regenerated from the cached LLM responses."
+                ) from e
+            except TypeError as e:
+                os.remove(dataset_file)
                 raise ValueError(
-                    "Arrow writer is complaining about the schema: likely all of your parsed rows were None and writer.write only wrote None objects."
+                    "Arrow writer is complaining about the types: likely your `parse_func` is not returning a valid list of rows (list of dictionaries). Please fix your `parse_func` to return valid rows and re-run. The dataset will be regenerated from the cached LLM responses."
                 ) from e
 
-        return Dataset.from_file(dataset_file)
+        try:
+            output_dataset = Dataset.from_file(dataset_file)
+        except Exception as e:
```

Comment:
another broad exception catching here here..

Reply:
i think this is ok since we are not doing anything here and just forwarding the error in a more understandable manner?

we can say something like, f"Failed to load dataset from {dataset_file}. This was likely corrupted by a failed previous run. You can delete this file and try again to see if the error is transient."

Reply:
Changed it to catch a specific error here `pyarrow.lib.ArrowInvalid` instead. 
Removing the file is better IMO since this happens at the top of the logic - the user might have fixed their `parse_func` from last time, but not deleted the file  (we shouldn't expect or encourage users to touch the cache).